src/test/shared/awsFiletypes.test.ts

```diff
@@ -0,0 +1,93 @@
+/*!
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+import * as assert from 'assert'
+import * as path from 'path'
+import globals from '../../shared/extensionGlobals'
+import * as sysutil from '../../shared/systemUtilities'
+import * as testUtil from '../testUtil'
+import * as workspaceUtils from '../../shared/utilities/workspaceUtils'
+import { toArrayAsync } from '../../shared/utilities/collectionUtils'
+import { waitUntil } from '../../shared/utilities/timeoutUtils'
+
+describe('awsFiletypes', function () {
+    let awsConfigUri: vscode.Uri | undefined
+    let cfnUri: vscode.Uri | undefined
+
+    beforeEach(async function () {
+        globals.telemetry.clearRecords()
+        testUtil.closeAllEditors()
+
+        // Create a dummy file in ~/.aws on the system.
+        // Note: We consider _any_ file in ~/.aws to be an "AWS config" file,
+        // so this will trigger "file_editAwsFile" telemetry.
+        const awsConfigFile = path.join(sysutil.SystemUtilities.getHomeDirectory(), '.aws/test.awstoolkit')
+        awsConfigUri = vscode.Uri.file(awsConfigFile)
+        testUtil.toFile('Test file from the aws-toolkit-vscode test suite.', awsConfigFile)
+
+        const cfnFile = workspaceUtils.tryGetAbsolutePath(
+            vscode.workspace.workspaceFolders?.[0],
+            'python3.7-plain-sam-app/template.yaml'
+        )
+        cfnUri = vscode.Uri.file(cfnFile)
+    })
+
+    after(async function () {
+        globals.telemetry.clearRecords()
+        testUtil.closeAllEditors()
+    })
+
+    it('emit telemetry when opened by user', async function () {
+        await globals.templateRegistry.addItemToRegistry(cfnUri!)
+        await vscode.commands.executeCommand('vscode.open', cfnUri)
+        await vscode.commands.executeCommand('vscode.open', awsConfigUri)
```

Comment:
This tests the actual implementation. Then we just use `globals.telemetry.findIter()` to check what was emitted.